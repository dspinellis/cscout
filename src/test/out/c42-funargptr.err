Processing workspace TestWS
Processing directory test/c
Entering directory test/c
Processing project Prj1
Processing file c42-funargptr.c
Done processing file c42-funargptr.c
Done processing project Prj1
Processing project Prj2
Processing file c42-funargptr.c
Done processing file c42-funargptr.c
Processing file prj2.c
Done processing file prj2.c
Done processing project Prj2
Exiting directory test/c
Done processing directory test/c
Done processing workspace TestWS
Processing identifiers
3%6%9%12%16%19%22%25%29%32%35%38%41%45%48%51%54%58%61%64%67%70%74%77%80%83%87%90%93%96%100%
