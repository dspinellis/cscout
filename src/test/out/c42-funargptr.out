Table: Ids
EID	NAME	READONLY	UNDEFMACRO	MACRO	FUNMACRO	MACROARG	CPPCONST	CPPSTRVAL	DEFCCONSTVAL	NOTDEFCCONSTVAL	EXPCCONSTVAL	NOTEXPCCONSTVAL	ORDINARY	SUETAG	SUMEMBER	LABEL	TYPEDEF	ENUM	YACC	FUN	CSCOPE	LSCOPE	UNUSED
79	simple_function	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	0
91	MAXDIGIT	0	0	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	1
309	d	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0
321	x	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	1	1
421	main	1	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	0
541	qqq	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	1	1
619	f	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	1
679	fa	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0
789	fx	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	1
821	foo	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	1
869	fxa	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	1
969	fb	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0
1089	rfa	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	1
1209	lb	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0
2079	f2	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	1
2149	f2a	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0
2279	rf2a	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	1
2409	rf2b	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	1
2779	f4a	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	1	1
2879	f3	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	1	1
2949	f3a	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0
2975	__DATE__	1	0	1	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	1
3069	f4	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	0	0	0
3139	f4a	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0
3295	__TIME__	1	0	1	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	1
3575	__FILE__	1	0	1	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	1
3865	__LINE__	1	0	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	1
4055	__STDC__	1	0	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	1
5355	_cscout_dummy1	1	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	1	1	0	0
Table: Tokens
FID	FOFFSET	EID
2	297	2975
2	329	3295
2	357	3575
2	386	3865
2	405	4055
2	471	421
2	535	5355
2	558	5355
4	7	79
4	30	309
4	42	309
4	61	619
4	67	679
4	78	789
4	86	869
4	96	969
4	108	1089
4	120	1209
4	126	1209
4	139	679
4	145	969
4	158	79
4	207	2079
4	214	2149
4	227	2279
4	240	2409
4	249	2149
4	277	2779
4	287	2879
4	294	2949
4	306	3069
4	313	3139
4	330	3139
4	350	3069
4	353	2949
5	8	91
5	31	321
5	41	421
5	53	541
5	81	821
Table: Rest
FID	FOFFSET	CODE
2	287	\u0000a\u0000a#define 
2	305	  
2	320	\u0000a#define 
2	337	 
2	348	\u0000a#define 
2	365	 
2	377	\u0000a#define 
2	394	 1\u0000a#define 
2	413	 1\u0000a\u0000a
2	466	\u0000aint 
2	475	();\u0000a
2	522	\u0000astatic void 
2	549	(void) { 
2	572	(); }\u0000a
3	152	\u0000a\u0000a\u0000a#pragma includepath 
3	195	\u0000a
3	239	\u0000astatic void _cscout_dummy2(void) { _cscout_dummy2(); }\u0000a
4	0	double 
4	22	(double 
4	31	) { return 
4	43	 + 1; }\u0000a\u0000adouble (*
4	62	(int 
4	69	, void (*
4	80	)(int 
4	89	), int 
4	98	))(double 
4	111	)\u0000a{\u0000a	int 
4	122	;\u0000a\u0000a	
4	128	 = 3;\u0000a	if (
4	141	 && 
4	147	)\u0000a		return 
4	173	;\u0000a	else\u0000a		return 0;\u0000a}\u0000a\u0000adouble (*(*
4	209	(int 
4	217	))(double 
4	231	))(float 
4	244	)\u0000a{\u0000a	
4	252	--;\u0000a	return 0;\u0000a}\u0000a\u0000adouble 
4	280	;\u0000a\u0000aint 
4	289	(int 
4	297	)\u0000a{\u0000a	int 
4	308	(int 
4	316	)\u0000a	{\u0000a		return 
4	333	 + 2;\u0000a	}\u0000a	return 
4	352	(
4	356	);\u0000a}\u0000a
5	0	#define 
5	16	 11\u0000aextern int 
5	32	;\u0000aextern 
5	45	();\u0000aint 
5	56	;\u0000a\u0000a
5	80	\u0000a
5	84	() {\u0000a	
5	107	}\u0000a\u0000a
Table: Projects
PID	NAME
23	unspecified
24	Prj1
25	Prj2
Table: IdProj
EID	PID
79	24
309	24
421	24
619	24
679	24
789	24
869	24
969	24
1089	24
1209	24
2079	24
2149	24
2279	24
2409	24
2779	24
2879	24
2949	24
2975	24
3069	24
3139	24
3295	24
3575	24
3865	24
4055	24
5355	24
79	25
91	25
309	25
321	25
421	25
541	25
619	25
679	25
789	25
821	25
869	25
969	25
1089	25
1209	25
2079	25
2149	25
2279	25
2409	25
2779	25
2879	25
2949	25
2975	25
3069	25
3139	25
3295	25
3575	25
3865	25
4055	25
5355	25
Table: Files
FID	NAME	RO
2	host-defs.h	1
3	host-incs.h	1
4	c42-funargptr.c	0
5	prj2.c	0
Table: Filemetrics
FID	PRECPP	NCHAR	NCCOMMENT	NSPACE	NLCOMMENT	NBCOMMENT	NLINE	MAXLINELEN	MAXSTMTLEN	MAXSTMTNEST	MAXBRACENEST	MAXBRACKNEST	BRACENEST	BRACKNEST	NULINE	NPPDIRECTIVE	NPPCOND	NPPFMACRO	NPPOMACRO	NTOKEN	NSTMT	NOP	NUOP	NNCONST	NCLIT	NSTRING	NPPCONCATOP	NPPSTRINGOP	NIF	NELSE	NSWITCH	NCASE	NDEFAULT	NBREAK	NFOR	NWHILE	NDO	NCONTINUE	NGOTO	NRETURN	NASM	NTYPEOF	NPID	NFID	NMID	NID	NUPID	NUFID	NUMID	NUID	NLABEL	NMACROEXPANDTOKEN	NCOPIES	NINCFILE	NPFUNCTION	NFFUNCTION	NPVAR	NFVAR	NAGGREGATE	NAMEMBER	NENUM	NEMEMBER
2	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	16	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	18	2	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	NULL	NULL	0	1	0	0	0	0	0	0
2	1	548	367	29	0	3	22	61	47	NULL	1	1	0	0	0	5	0	0	5	37	2	0	0	2	0	3	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	2	5	8	1	1	5	7	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
3	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	15	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	1	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	NULL	NULL	0	0	0	0	0	0	0	0
3	1	243	187	13	0	2	13	48	28	NULL	1	1	0	0	0	1	0	0	0	16	1	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
4	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	44	1	2	3	NULL	NULL	NULL	NULL	NULL	NULL	NULL	120	11	9	0	5	0	0	NULL	NULL	1	1	0	0	0	0	0	0	0	0	0	6	0	0	12	0	NULL	54	5	0	NULL	18	0	NULL	NULL	NULL	4	0	1	0	0	0	0	0
4	1	294	0	83	0	0	29	47	46	NULL	2	3	0	0	0	0	0	0	0	119	11	9	0	5	0	0	0	0	1	1	0	0	0	0	0	0	0	0	0	6	0	0	6	0	0	27	5	0	0	18	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
5	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	7	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	18	3	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	4	0	NULL	4	4	0	NULL	4	0	NULL	NULL	NULL	1	0	2	0	0	0	0	0
5	1	95	32	17	2	0	11	20	13	NULL	1	1	0	0	0	1	0	0	1	21	3	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	4	0	1	5	4	0	1	5	0	0	1	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL
Table: FileProj
FID	PID
2	24
3	24
4	24
1	25
2	25
3	25
4	25
5	25
Table: LineProj
FID	PID	LNUM
2	24	1
2	24	2
2	24	3
2	24	4
2	24	5
2	24	6
2	24	7
2	24	8
2	24	9
2	24	10
2	24	11
2	24	12
2	24	13
2	24	14
2	24	15
2	24	16
2	24	17
2	24	18
2	24	19
2	24	20
2	24	21
2	24	22
3	24	1
3	24	2
3	24	3
3	24	4
3	24	5
3	24	6
3	24	7
3	24	8
3	24	9
3	24	10
3	24	11
3	24	12
3	24	13
4	24	1
4	24	2
4	24	3
4	24	4
4	24	5
4	24	6
4	24	7
4	24	8
4	24	9
4	24	10
4	24	11
4	24	12
4	24	13
4	24	14
4	24	15
4	24	16
4	24	17
4	24	18
4	24	19
4	24	20
4	24	21
4	24	22
4	24	23
4	24	24
4	24	25
4	24	26
4	24	27
4	24	28
4	24	29
2	25	1
2	25	2
2	25	3
2	25	4
2	25	5
2	25	6
2	25	7
2	25	8
2	25	9
2	25	10
2	25	11
2	25	12
2	25	13
2	25	14
2	25	15
2	25	16
2	25	17
2	25	18
2	25	19
2	25	20
2	25	21
2	25	22
3	25	1
3	25	2
3	25	3
3	25	4
3	25	5
3	25	6
3	25	7
3	25	8
3	25	9
3	25	10
3	25	11
3	25	12
3	25	13
4	25	1
4	25	2
4	25	3
4	25	4
4	25	5
4	25	6
4	25	7
4	25	8
4	25	9
4	25	10
4	25	11
4	25	12
4	25	13
4	25	14
4	25	15
4	25	16
4	25	17
4	25	18
4	25	19
4	25	20
4	25	21
4	25	22
4	25	23
4	25	24
4	25	25
4	25	26
4	25	27
4	25	28
4	25	29
5	25	1
5	25	2
5	25	3
5	25	4
5	25	5
5	25	6
5	25	7
5	25	8
5	25	9
5	25	10
5	25	11
Table: LinePos
FID	FOFFSET	LNUM
2	0	1
2	4	2
2	51	3
2	54	4
2	100	5
2	119	6
2	122	7
2	184	8
2	231	9
2	234	10
2	284	11
2	288	12
2	289	13
2	321	14
2	349	15
2	378	16
2	397	17
2	416	18
2	417	19
2	467	20
2	479	21
2	523	22
3	0	1
3	4	2
3	51	3
3	54	4
3	78	5
3	97	6
3	100	7
3	149	8
3	153	9
3	154	10
3	155	11
3	196	12
3	240	13
4	0	1
4	51	2
4	52	3
4	113	4
4	115	5
4	124	6
4	125	7
4	134	8
4	149	9
4	175	10
4	181	11
4	193	12
4	195	13
4	196	14
4	246	15
4	248	16
4	256	17
4	267	18
4	269	19
4	270	20
4	282	21
4	283	22
4	299	23
4	301	24
4	318	25
4	321	26
4	339	27
4	342	28
4	359	29
5	0	1
5	20	2
5	34	3
5	49	4
5	58	5
5	59	6
5	80	7
5	81	8
5	89	9
5	107	10
5	109	11
Table: Definers
PID	CUID	BASEFILEID	DEFINERID
25	5	5	2
Table: Includers
PID	CUID	BASEFILEID	INCLUDERID
24	2	2	1
24	4	3	1
24	4	4	1
25	2	2	1
25	2	2	1
25	4	3	1
25	4	4	1
25	5	3	1
25	5	5	1
Table: Providers
PID	CUID	PROVIDERID
24	2	2
24	4	4
25	2	2
25	2	2
25	4	4
25	5	5
Table: IncTriggers
PID	CUID	BASEFILEID	DEFINERID	FOFFSET	LEN
25	5	5	2	471	4
Table: Functions
ID	NAME	ISMACRO	DEFINED	DECLARED	FILESCOPED	FID	FOFFSET	FANIN
79	simple_function	0	1	1	0	4	7	1
619	f	0	1	1	0	4	61	0
821	foo	0	1	1	0	5	81	0
2079	f2	0	1	1	0	4	207	0
2879	f3	0	1	1	0	4	287	0
3069	f4	0	1	1	0	4	306	1
4715	main	0	0	1	0	2	471	0
5355	_cscout_dummy1	0	1	1	1	2	535	1
Table: FunctionDefs
FUNCTIONID	FIDBEGIN	FOFFSETBEGIN	FIDEND	FOFFSETEND
79	4	34	4	50
619	4	114	4	194
821	5	88	5	108
2079	4	247	4	268
2879	4	300	4	360
3069	4	320	4	341
5355	2	557	2	577
Table: FunctionMetrics
FUNCTIONID	PRECPP	NCHAR	NCCOMMENT	NSPACE	NLCOMMENT	NBCOMMENT	NLINE	MAXLINELEN	MAXSTMTLEN	MAXSTMTNEST	MAXBRACENEST	MAXBRACKNEST	BRACENEST	BRACKNEST	NULINE	NPPDIRECTIVE	NPPCOND	NPPFMACRO	NPPOMACRO	NTOKEN	NSTMT	NOP	NUOP	NNCONST	NCLIT	NSTRING	NPPCONCATOP	NPPSTRINGOP	NIF	NELSE	NSWITCH	NCASE	NDEFAULT	NBREAK	NFOR	NWHILE	NDO	NCONTINUE	NGOTO	NRETURN	NASM	NTYPEOF	NPID	NFID	NMID	NID	NUPID	NUFID	NUMID	NUID	NLABEL	NMACROEXPANDTOKEN	NGNSOC	NMPARAM	NFPARAM	NEPARAM	FANIN	FANOUT	CCYCL1	CCYCL2	CCYCL3	CSTRUC	CHAL	IFLOW
79	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	9	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	6	1	1	1	1	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	NULL	1	0	0	NULL	1	0	NULL	11	NULL	1	NULL	1	0	1	1	1	0.0	4.75489	0.0
79	1	13	0	6	0	0	1	12	9	NULL	0	0	0	0	0	0	0	0	0	6	1	1	1	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	1	0	0	0	1	0	0	NULL	0	NULL	0	1	0	1	1	1	0.0	4.75489	0.0
619	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	15	1	0	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	21	5	2	2	2	0	0	NULL	NULL	1	1	0	0	0	0	0	0	0	0	0	2	0	0	1	0	NULL	5	1	0	NULL	4	0	NULL	13	NULL	3	NULL	0	1	2	3	3	0.0	27.0	0.0
619	1	70	0	25	0	0	9	21	17	NULL	0	1	0	0	0	0	0	0	0	21	5	2	2	2	0	0	0	0	1	1	0	0	0	0	0	0	0	0	0	2	0	0	1	0	0	5	1	0	0	4	0	0	NULL	0	NULL	0	0	1	2	3	3	0.0	27.0	0.0
821	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	1	0	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	0	0	NULL	0	0	NULL	22	NULL	0	NULL	0	0	1	1	1	0.0	0.0	0.0
821	1	21	14	3	1	0	3	17	0	NULL	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	NULL	0	NULL	0	0	0	1	1	1	0.0	0.0	0.0
2079	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	5	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	7	2	1	1	1	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	NULL	1	0	0	NULL	1	0	NULL	15	NULL	1	NULL	0	0	1	1	1	0.0	4.75489	0.0
2079	1	18	0	7	0	0	4	8	6	NULL	0	0	0	0	0	0	0	0	0	7	2	1	1	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	1	0	0	0	1	0	0	NULL	0	NULL	0	0	0	1	1	1	0.0	4.75489	0.0
2879	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	19	0	1	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	14	1	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	NULL	4	0	0	NULL	3	0	NULL	19	NULL	1	NULL	0	1	1	1	1	0.0	6.33985	0.0
2879	1	34	0	10	0	0	4	15	22	NULL	1	1	1	0	0	0	0	0	0	14	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	4	0	0	0	3	0	0	NULL	0	NULL	0	0	1	1	1	1	0.0	6.33985	0.0
3069	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	9	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	6	1	1	1	1	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	NULL	1	0	0	NULL	1	0	NULL	20	NULL	1	NULL	1	0	1	1	1	0.0	4.75489	0.0
3069	1	18	0	9	0	0	3	13	10	NULL	0	0	0	0	0	0	0	0	0	6	1	1	1	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	0	0	0	0	1	0	0	0	1	0	0	NULL	0	NULL	0	1	0	1	1	1	0.0	4.75489	0.0
5355	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	5	0	0	1	NULL	NULL	NULL	NULL	NULL	NULL	NULL	5	1	0	0	0	0	0	NULL	NULL	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	NULL	1	0	1	NULL	1	0	NULL	14	NULL	0	NULL	1	1	1	1	1	1.0	0.0	1.0
5355	1	22	0	3	0	0	1	21	5	NULL	0	1	0	0	0	0	0	0	0	5	1	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	0	1	0	1	0	1	0	1	0	0	NULL	0	NULL	0	1	1	1	1	1	1.0	0.0	1.0
Table: FunctionId
FUNCTIONID	ORDINAL	FID	FOFFSET	LEN
79	0	4	7	15
619	0	4	61	1
821	0	5	81	3
2079	0	4	207	2
2879	0	4	287	2
3069	0	4	306	2
4715	0	2	471	4
5355	0	2	535	14
Table: Fcalls
SOURCEID	DESTID
619	79
2879	3069
5355	5355
Done
